src/components/Identity/createMigration.js

```diff
@@ -14,8 +14,11 @@ export default (imsOrgId, migrateIds) => {
       if (migrateIds) {
         const amcvCookieValue = cookieJar.get(`AMCV_${imsOrgId}`);
         if (amcvCookieValue) {
-          const reg = /MCMID\|(\d+)\|/;
-          [, ecid] = amcvCookieValue.match(reg);
+          const reg = /(^|\|)MCMID\|(\d+)($|\|)/;
```

Comment:
I came across an amcv cookie value that was just `MCMID|123` where `123` was a long number. The regex that we had would only match if there was a pipe after the number, which there wasn't in this case. I think this was an unintentional bug or maybe I somehow had an unusual amcv cookie in my sandbox, but @jileeshs should double-check my work here.

Reply:
I think it is possible to have an AMCV cookie with `MCMID|546736y47u239057230958` with no `|` at the end.